### ISSUE TITLE
Set parameter

### ISSUE THREAD

**Issue by jherven** [dmitry-zhuravlev/rest-runner-teamcity-plugin#1]
Thanks for making a usable plugin.

I have a question/feature request (depending on the answer).

Is it possible to set a parameter during the reponse analyze that can be used in later build steps?
If have tried a few variants, for example
println "##teamcity[setParameter name='system.myParameter' 'someValue']"
but the values doesn't stick. (Just empty in later steps)

Is this something that's not possible or am I just doing it wrong?

**Comment by dmitry-zhuravlev** [comment]
Hi @jherven Thank you for feedback!
Where did you try to print service message `##teamcity[setParameter name='system.myParameter' 'someValue']`?
I think that service message mechanism is undependant from any runners and if you print it to the standard output stream of the build it should works.

But if you want that response analyzer set some property depending on the analyze result it is probably like feature request. Can you summarize what kind of parameter should be set by response analyzer and what is the common use case for it?

**Comment by jherven** [comment]
A bit simplified my scenario looks like this. I have two websites that can either be in a active or passive state. A simple html file on each site tells which state currently applies.

What I tried to accomplish using your plugin was to detect which site is currently passive, set a parameter to this site identifier and use that later in my deploy step. 

So basically what I did was to start with a **REST Runner** step:
**Endpoint:** GET http://mysite1.com
**Groovy script:**
if("$response" == "passive")
print ##teamcity[setParameter name='system.myParameter' 'site1']
else
print ##teamcity[setParameter name='system.myParameter' 'site2']

I was hoping the script should write to the standard output stream, or that there was some alternative function I didn't found. Maybe this it outside the scope for this plugin and there are other better ways to solve this, but I actually haven't found any until I came across this plugin which felt very neat and straight forward. 

In either way, thanks for taking the time to answer

**Comment by dmitry-zhuravlev** [comment]
Hm... Very interesting use case...
When I first designed this Groovy script evaluation, it was just to evaluate true/false of some condition met. I didn't even think that service messages will be also usefull inside such script. 

As I mention previously `print ...` will be definitelly not work because script evaluated with separated script engine. 
Probably I can add `buildLogger` instance to that script and you can just call:
`buildLogger.message('##teamcity[setParameter name='system.myParameter' 'site1']')` 

@jherven Would it help you?

**Comment by jherven** [comment]
@dmitry-zhuravlev That would be a perfect solution

**Comment by dmitry-zhuravlev** [comment]
@jherven Just released 1.1 version with such update. 
Have a try and let me know if it works for you.

**Comment by jherven** [comment]
Worked like a charm.
Very much appreciated you could implement this so fast.
